python/build-wheel-macos-arm64.sh

```diff
@@ -22,7 +22,6 @@ if [[ -n "${SKIP_DEP_RES}" ]]; then
 
   curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
   curl -o- https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh | bash
-  source ~/.bash_profile
   conda init bash
```

Comment:
I feels like if we move these 2 lines out of the if statement, we can avoid a lot of the new branches being added to this script?

Reply:
Let's clean this up in a follow-up. We should generally look if we can improve these scripts.

For now I want to avoid changing too much of the existing CI scripts before the branch cut